test/e2e/certificate-available.spec.ts

```diff
@@ -0,0 +1,90 @@
+import { test, expect } from '@playwright/test';
+import dayjs from 'dayjs';
+import { prisma } from '../../app/db.server';
+import { loggedInAsUser } from './utils';
+
+import type { Certificate } from '@prisma/client';
+
+test.describe('Certificate Page', () => {
+  loggedInAsUser();
+
+  test.beforeAll(async () => {
+    const data: Pick<
+      Certificate,
+      | 'username'
+      | 'domain'
+      | 'orderUrl'
+      | 'privateKey'
+      | 'certificate'
+      | 'chain'
+      | 'validFrom'
+      | 'validTo'
+      | 'status'
+    > = {
+      username: 'user1',
+      domain: 'user1.starchart.com',
+      orderUrl: `${new Date().toISOString()}.example.com`,
+      privateKey:
+        '-----BEGIN CERTIFICATE-----ApfFCv0O65TYkp5jEWSlU8PhKYD43nXA=-----END CERTIFICATE-----',
+      certificate:
+        '-----BEGIN CERTIFICATE-----BpfFCv0OuF8AujEWv0Okp5jEWSlAuD43=-----END CERTIFICATE-----',
+      chain:
+        '-----BEGIN CERTIFICATE-----CjEWSlU8PhKYTYWSlU8hKYTYkp5jewDW=-----END CERTIFICATE-----',
+      validFrom: new Date(),
+      validTo: dayjs().add(1, 'day').toDate(),
+      status: 'issued',
+    };
+    await prisma.certificate.create({ data });
+  });
+
+  test.beforeEach(async ({ page }) => {
+    await page.goto('/certificate');
+  });
+
+  test.afterAll(async () => {
+    await prisma.certificate.deleteMany({});
+  });
+
+  test('Request a Certificate', async ({ page }) => {
```

Comment:
This test description doesn't describe what you're doing below.

Also, you are doing too many things in this one test.  Separate copy from download